### ISSUE TITLE
Possibility of including new image data via DIY capture of Google Street View imagery

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#2818]
Since the launch of Project Sidewalk (and even before when working on precursor projects), people have asked if we could add in features to add in their own geo-tagged smartphone photos of areas. My response was always that this was out-of-scope for our focus and relied on on-the-ground community contributions that would be sparse, at best.

However, in discussions with Jesus Rodriguez in La Piedad, we've been talking about a different data refresh approach, which is to have people upload photos to Google Street View, which would then be (hopefully) automatically used by our tool. The key concern in La Piedad is data age—many of the GSV panos seem to be from 2010. So, they asked about whether their team could do their own DIY streetview upload. There are lots of blog posts and official Google posts about this, including a recent post that talks about using modern Android/iPhone phones as capture devices (rather than specialized 360 pano cameras).

> With our updated [Street View app](https://play.google.com/store/apps/details?id=com.google.android.street) on Android, it’s now easier than ever to collect your own Street View imagery and put it in the right place on Google Maps. Using our new connected photos tool in the app, you can record a series of connected images as you move down a street or path. 
>
> These images are captured using [ARCore](https://developers.google.com/ar), the same augmented reality technology we use to produce experiences like [Live View](https://blog.google/products/maps/new-sense-direction-live-view/). After you record your images and publish them via the Street View app, we automatically rotate, position and create a series of connected photos. We then place those connected images in the right place on Google Maps, so your new Street View can be found in the exact location where it was taken for others to see and explore.
>
> Before this feature, you would typically need special [360-degree cameras](https://www.google.com/streetview/contacts-tools/) to capture and publish Street View imagery. Some equipment you could even attach to the roof of your car, but at the cost of thousands of dollars; that’s out of the realm for many.

See also:

- https://blog.google/products/maps/anyone-can-share-their-world-with-street-view/
- https://support.google.com/maps/answer/7011737?hl=en
- https://www.google.com/streetview/contacts-tools/

**Comment by jonfroehlich** [comment]
I just tried this myself over by the university in an intersection with brand new curb ramps (installed last month). I downloaded the [Google Street View app](https://apps.apple.com/us/app/google-street-view/id904418768) on my iPhone and found the overall process quite easy—with a nice UI and step-by-step instructions in the app. I used an iPhone 13 (not sure what models of Android phones and iPhones are supported).

I uploaded the imagery this morning (attached) but not sure when/if it will be approved.

# Final Image (Auto-stitched by App):
![output](https://user-images.githubusercontent.com/1621749/158648336-d84cdbb3-4cd9-414f-bb23-a8897b6da158.jpg)

# Some screenshots
![IMG_4713](https://user-images.githubusercontent.com/1621749/158648325-ff0d6e32-abea-4105-8457-65e63fb79913.PNG)

![IMG_4710](https://user-images.githubusercontent.com/1621749/158648306-70d55816-ff96-4ca7-a177-d5f2d41b1b5b.PNG)

![IMG_4709](https://user-images.githubusercontent.com/1621749/158648292-e82666c8-58fc-46e9-a902-f7ac2952b4ef.PNG)

**Comment by jonfroehlich** [comment]
These appear to now all be [published here](https://www.google.com/maps/contrib/110326759916505068616/photos/@47.6626596,-122.2881927,3a,75y,229h,90t/data=!3m7!1e1!3m5!1sAF1QipOGdZRBQTxEB9XiEmJquWYp8KF9iySSNgF1iaVj!2e10!6shttps:%2F%2Flh5.googleusercontent.com%2Fp%2FAF1QipOGdZRBQTxEB9XiEmJquWYp8KF9iySSNgF1iaVj%3Dw365-h260-k-no-pi-0-ya230.99998-ro0-fo100!7i4608!8i2304!4m3!8m2!3m1!1e1), however, I can't seem to access them via normal Google Street View.

**Comment by jonfroehlich** [comment]
I'm curious if we could access them via the GSV API.

**Comment by jonfroehlich** [comment]
This is related to a recent comment from the La Piedad folks, who said:

> Hola buen día, por nosotros encantados, me preocupa un poco la desactualización de las imágenes. El motivo de realizar el proyecto es tener el apoyo de la gente de La Piedad, pero su apoyo sea desde sus casas y no tener que ir a campo, si vamos a campo el proyecto pierde su esencia.
> Estamos solicitando el apoyo del CIGA UNAM [https://ciga.unam.mx](https://t.co/qbxW6ssasN) ellos tienen una cámara 360° y queremos invitarlos al proyecto y así poder actualizar la información del street view.
> Me gustaría saber su opinión, gracias

English translation:
> Hello good day, we are delighted, I am a little concerned about the outdated images. The reason for carrying out the project is to have the support of the people of La Piedad, but their support is from their homes and not having to go to the field, if we go to the field the project loses its essence.
> We are requesting the support of CIGA UNAM [https://ciga.unam.mx](https://t.co/qbxW6ssasN) they have a 360° camera and we want to invite them to the project so we can update the street view information.
> I would like to know your opinion, thank you

**Comment by jonfroehlich** [comment]
Spoke to Liga again as we open new neighborhoods in La Piedad. They said:

> Thanks Jon, I think this is a good opportunity to pilot a solution to this issue that will appear in the rest of the cities. The ideal would be to have an app that allows us to create labels directly from the site. But that I guess is going to require more work, maybe just adding an option in the "label map" to create them directly from the aerial view, and not from the street view. What do you think?

So, can we make a quick Android/iPhone app to let people to upload tagged photos of problems that we add to LabelMap!

**Comment by jonfroehlich** [comment]
Some updates!

![image](https://user-images.githubusercontent.com/1621749/202532808-d62ee678-2d33-4809-ba85-370d45dd04bb.png)

We actually found that you can see my 360 photo spheres (taken from my iPhone) if you zoom way in:

![image](https://user-images.githubusercontent.com/1621749/202533023-f98f4d9d-6b56-4617-93d8-6733297b147e.png)

![image](https://user-images.githubusercontent.com/1621749/202533157-dadc0052-4b0a-48f0-895b-32605ca62742.png)

And then Mikey found that he can see it with our tool!

**Comment by jonfroehlich** [comment]
Mikey was able to explore my DIY contributions with Project Sidewalk:

![image](https://user-images.githubusercontent.com/1621749/202533627-79208ce5-7bd3-4359-a77e-013e9cc3a224.png)

One small lil hic up was that we also discovered the pano id went off the page!

![image](https://user-images.githubusercontent.com/1621749/202533692-667afcfd-76bf-4a4a-bd63-2500f872b466.png)

The View in Google Street View brings us here: https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=47.659148599999995%2C-122.2892861&heading=90&pitch=-10

**Comment by misaugstad** [comment]
The issue with using it on Project Sidewalk, though, was that it isn't connected/linked to nearby panos (you can see that there are no arrows linking to other panos in the screenshot). I only ended up at that pano because I was dropped directly into it when loading the audit page. We could also get there by clicking the 'stuck' button.

Hopefully if we add a string of panos, it will get connected to StreetView, as noted in the documentation that Jon showed above. Though this can be difficult in high traffic areas...

**Comment by jonfroehlich** [comment]
Exactly. The next step of our investigation is for me (and possibly the kiddos) to take multiple photos to see if we can get a "photo path" of connected photos (just like GSV cars offer)
